cmd/diag/downloader/diag_downloader.go

```diff
@@ -1,77 +1,324 @@
 package downloader
 
 import (
-	"encoding/json"
 	"fmt"
 
+	"github.com/jedib0t/go-pretty/v6/table"
 	"github.com/ledgerwatch/erigon-lib/common"
 	"github.com/ledgerwatch/erigon-lib/diagnostics"
 	"github.com/ledgerwatch/erigon/cmd/diag/flags"
 	"github.com/ledgerwatch/erigon/cmd/diag/util"
 	"github.com/urfave/cli/v2"
 )
 
+var (
+	SegmentsFilterFlag = cli.StringFlag{
+		Name:     "downloader.segments.filter",
+		Aliases:  []string{"dsf"},
+		Usage:    "Filter segments list [all|active|inactive|downloaded|queued], dafault value is all",
+		Required: false,
+		Value:    "all",
+	}
+
+	SenmentNameFlag = cli.StringFlag{
+		Name:     "downloader.segment.name",
+		Aliases:  []string{"dsn"},
+		Usage:    "Segment name to print details about.",
+		Required: true,
+		Value:    "",
+	}
+)
+
 var Command = cli.Command{
-	Action:    print,
+	Action:    printDownloadStatus,
 	Name:      "downloader",
 	Aliases:   []string{"dl"},
-	Usage:     "print snapshot download stats",
+	Usage:     "Print snapshot download status",
 	ArgsUsage: "",
 	Flags: []cli.Flag{
 		&flags.DebugURLFlag,
 		&flags.OutputFlag,
 	},
+	Subcommands: []*cli.Command{
+		{
+			Name:      "segments",
+			Aliases:   []string{"segs"},
+			Action:    printSegmentsList,
+			Usage:     "print current stage",
+			ArgsUsage: "",
+			Flags: []cli.Flag{
+				&flags.DebugURLFlag,
+				&flags.OutputFlag,
+				&SegmentsFilterFlag,
+			},
+		},
+		{
+			Name:      "segment",
```

Comment:
is that correct? `segment(s)`

Reply:
segments - to print all segments for snapshot
segment - to print one with specifying name

If it confusing please suggest what will be better.

Reply:
all segments for all snap files or for one file? Pls add that distinction into command description